python/lsst/pipe/tasks/compute_object_epochs.py

```diff
@@ -0,0 +1,133 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from astropy.table import Table
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import numpy as np
+
+
+class ComputeObjectEpochsConnections(
+    pipeBase.PipelineTaskConnections,
+    dimensions=("tract", "skymap"),
+):
+    objectCat = pipeBase.connectionTypes.Input(
+        doc="Multiband catalog of positions in each patch.",
+        name="deepCoadd_obj",
+        storageClass="ArrowAstropy",
+        dimensions=["skymap", "tract", "patch"],
+        multiple=True,
+        deferLoad=True,
+        deferGraphConstraint=True,
+    )
+
+    epochMap = pipeBase.connectionTypes.Input(
+        doc="Healsparse map of mean epoch of objectCat in each band.",
+        name="deepCoadd_epoch_map_mean",
+        storageClass="HealSparseMap",
+        dimensions=("skymap", "tract", "band"),
+        multiple=True,
+        deferLoad=True,
+    )
+
+    objectEpochs = pipeBase.connectionTypes.Output(
+        doc="Catalog of epochs for objectCat objects.",
+        name="object_epoch",
+        storageClass="ArrowAstropy",
+        dimensions=["skymap", "tract", "patch"],
+        multiple=True,
+    )
+
+
+class ComputeObjectEpochsConfig(
+    pipeBase.PipelineTaskConfig,
+    pipelineConnections=ComputeObjectEpochsConnections,
+):
+    bands = pexConfig.ListField(
+        doc=("Bands in objectCat to be combined with `objectCat_selectors` to build objectCat column names."),
+        dtype=str,
+        default=["u", "g", "r", "i", "z", "y"],
+    )
+
+
+class ComputeObjectEpochsTask(pipeBase.PipelineTask):
+    """Collect mean epochs for the observations that went into each objepipeBase.connectionTypes.
+
+    TODO: DM-46202, Remove this task once the object epochs are available
+    elsewhere.
+    """
+
+    ConfigClass = ComputeObjectEpochsConfig
+    _DefaultName = "computeObjectEpochs"
+
+    def getEpochs(self, cat, epochMapDict):
+        """Get mean epoch of the visits corresponding to object position.
+
+        Parameters
+        ----------
+        cat : `astropy.table.Table`
+            Catalog containing object positions.
+        epochMapDict: `dict` [`DeferredDatasetHandle`]
+            Dictionary of handles for healsparse maps containing the mean epoch
+            for positions in the reference catalog.
+
+        Returns
+        -------
+        epochDf = `astropy.table.Table`
+            Catalog with mean epoch of visits at each object position.
+        """
+        allEpochs = {}
+        for band in self.config.bands:
```

Comment:
Given how it's used, I think it's worth considering whether we should drop the `bands` config option.  Unlike other tasks that only take `band`-free inputs like the standardized object table, this task does take a property map with `band` in its data IDs, so there's not really any danger of that set of bands exploding to all of the bands in the data repository.  And it's always nice to have one fewer `bands` config to have to configure when setting up a new variant of a pipeline.